curated_transformers/tests/models/gpt_neox/test_decoder.py

```diff
@@ -1,103 +1,36 @@
+    assert_decoder_output_equals_hf(
+        GPTNeoXDecoder,
+        "trl-internal-testing/tiny-random-GPTNeoXForCausalLM",
+        torch_device,
     )
-    hf_model.to(torch_device)
-    hf_model.eval()
-
-    model = GPTNeoXDecoder.from_hf_hub(
-        name="trl-internal-testing/tiny-random-GPTNeoXForCausalLM", device=torch_device
-    )
-    model.eval()
-
-    torch.manual_seed(0)
-    X = torch.randint(0, hf_model.config.vocab_size, (2, 10), device=torch_device)
-
-    with torch.no_grad():
-        Y = model(X).last_hidden_layer_state
-        Y_hf = hf_model(X).last_hidden_state
-
-    torch_assertclose(Y, Y_hf)
-
-
-@pytest.mark.skipif(not has_hf_transformers, reason="requires huggingface transformers")
-@pytest.mark.parametrize("torch_device", TORCH_DEVICES)
-def test_decoder_with_cache(torch_device):
-    hf_model = transformers.AutoModel.from_pretrained(
-        "trl-internal-testing/tiny-random-GPTNeoXForCausalLM"
-    )
-    hf_model.to(torch_device)
-    hf_model.eval()
-
-    model = GPTNeoXDecoder.from_hf_hub(
-        name="trl-internal-testing/tiny-random-GPTNeoXForCausalLM", device=torch_device
-    )
-    model.eval()
-
-    torch.manual_seed(0)
-    X = torch.randint(0, hf_model.config.vocab_size, (2, 10), device=torch_device)
-    X_rest = torch.randint(0, hf_model.config.vocab_size, (2, 10), device=torch_device)
-
-    with torch.no_grad():
-        Y = model(X, store_cache=True)
-        Y_hf = hf_model(X, use_cache=True)
-        Y = model(X_rest, cache=Y.cache).last_hidden_layer_state
-        Y_hf = hf_model(X_rest, past_key_values=Y_hf.past_key_values).last_hidden_state
-
-    torch_assertclose(Y, Y_hf)
 
 
+@pytest.mark.slow
 @pytest.mark.skipif(not has_hf_transformers, reason="requires huggingface transformers")
+@pytest.mark.skipif(not has_torch_compile, reason="requires torch.compile")
 @pytest.mark.parametrize("torch_device", TORCH_DEVICES)
-def test_decoder_with_positions(torch_device):
-    hf_model = transformers.AutoModel.from_pretrained(
-        "trl-internal-testing/tiny-random-GPTNeoXForCausalLM"
-    )
-    hf_model.to(torch_device)
-    hf_model.eval()
-
-    model = GPTNeoXDecoder.from_hf_hub(
-        name="trl-internal-testing/tiny-random-GPTNeoXForCausalLM", device=torch_device
-    )
-    model.eval()
-
-    torch.manual_seed(0)
-    X = torch.randint(0, hf_model.config.vocab_size, (2, 10), device=torch_device)
-    # NOTE: this test may break in the future because we are relying on a
-    #       slicing bug, which results in more general support for positions
-    #       (like our implementation):
-    #
-    #       https://github.com/huggingface/transformers/blob/f924df3c7e5150317ef47754a31cebf2893570ce/src/transformers/models/gpt_neox/modeling_gpt_neox.py#L278
+def test_decoder_with_torch_compile(torch_device):
+    # NOTE: testing with cache/positions fails. All outputs seems ok until
+    # the concatenation of the vectors with/without rotary embeddings:
```

Comment:
Shall we (temporarily) add an `xfail` test case for that?

Reply:
I considered doing this, since it's a known failure. However, the problem is that if Torch fixes this issue, the test will suddenly pass (and thus fail), which is probably not nice.